pkg/function/service.go

```diff
@@ -132,6 +134,42 @@ func (fs *Service) MapFn(ctx context.Context, d *functionpb.DatumRequest) (*func
 	return datumList, nil
 }
 
+// MapStreamFn applies a function to each datum element and returns a stream.
+func (fs *Service) MapStreamFn(d *functionpb.DatumRequest, stream functionpb.UserDefinedFunction_MapStreamFnServer) error {
+	var hd = handlerDatum{
+		value:     d.GetValue(),
+		eventTime: d.GetEventTime().EventTime.AsTime(),
+		watermark: d.GetWatermark().Watermark.AsTime(),
+		metadata: handlerDatumMetadata{
+			id:           d.GetMetadata().GetId(),
+			numDelivered: d.GetMetadata().GetNumDelivered(),
+		},
+	}
+	ctx := stream.Context()
+	messageCh := make(chan Message)
```

Comment:
Do we expect the udf developer to close the channel in the end of their `handle()` function?

I think no matter whether the udf close the channel or not, we should support both cases (1. Channel is closed. 2. Function returned without closing the channel). Both of these cases can be considered as the udf function is done. And we should make sure the channel is closed if it's not done by the udf.

Thoughts?

Reply:
+1, we should support both.